network/network.go

```diff
@@ -698,10 +698,6 @@ func (n *network) getPeers(
 			continue
 		}
 
-		if trackedSubnets := peer.TrackedSubnets(); !trackedSubnets.Contains(subnetID) {
-			continue
-		}
```

Comment:
After going through the code - the one potential downside to this PR is:

If a subnet validator forgets to `--track-subnet` the subnet that it is validating, the nodes querying them will now need to timeout the request rather than immediately marking the request as failed.

I think that is sufficiently niche that it isn't a concern to merge this PR... But it could be something to keep in mind as a future optimization. This is not a change in any security assumptions, as byzantine nodes could always report that they are tracking the subnet but then just never respond to the request.